### PR TITLE
Document `nil`-able `old_*` states in update events.

### PR DESCRIPTION
**Pull request by jchristgit**
Felt this might be useful to have in the documentation.

### CODE REVIEW

File: lib/nostrum/consumer.ex [hunk 1]
```diff
@@ -66,8 +66,13 @@ defmodule Nostrum.Consumer do
 
   @type channel_create :: {:CHANNEL_CREATE, {Channel.t()}, WSState.t()}
   @type channel_delete :: {:CHANNEL_DELETE, {Channel.t()}, WSState.t()}
+  @typedoc """
+  Dispatched when a channel is updated.
+  `old_channel` will be `nil` when the pre-update channel could not be fetched from the cache.
```
Comment (Kraigie) [inline lib/nostrum/consumer.ex]: Could you add a new line between the two sentences? ExDoc uses the first line to generate summaries, so we should try to keep it away from the rest of the provided info just for clarity.

File: lib/nostrum/consumer.ex [hunk 2]
```diff
@@ -99,19 +104,27 @@ defmodule Nostrum.Consumer do
   @type guild_member_remove ::
           {:GUILD_MEMBER_REMOVE,
            {guild_id :: integer, old_member :: Nostrum.Struct.Guild.Member.t()}, WSState.t()}
+  @typedoc """
+  Dispatched when a guild member is updated.
+  `old_member` will be `nil` when the pre-update member could not be fetched from the cache.
```
Comment (Kraigie) [inline lib/nostrum/consumer.ex]: Could you add a new line between the two sentences? ExDoc uses the first line to generate summaries, so we should try to keep it away from the rest of the provided info just for clarity.

File: lib/nostrum/consumer.ex [hunk 3]
```diff
@@ -99,19 +104,27 @@ defmodule Nostrum.Consumer do
...
             new_member :: Nostrum.Struct.Guild.Member.t()}, WSState.t()}
   @type guild_role_create ::
           {:GUILD_ROLE_CREATE, {guild_id :: integer, new_role :: Nostrum.Struct.Guild.Role.t()},
            WSState.t()}
   @type guild_role_delete ::
           {:GUILD_ROLE_DELETE, {guild_id :: integer, old_role :: Nostrum.Struct.Guild.Role.t()},
            WSState.t()}
+  @typedoc """
+  Dispatched when a role on a guild is updated.
+  `old_role` will be `nil` when the pre-update role could not be fetched from the cache.
```
Comment (Kraigie) [inline lib/nostrum/consumer.ex]: Could you add a new line between the two sentences? ExDoc uses the first line to generate summaries, so we should try to keep it away from the rest of the provided info just for clarity.

File: lib/nostrum/consumer.ex [hunk 4]
```diff
@@ -123,16 +136,24 @@ defmodule Nostrum.Consumer do
   @type message_reaction_remove :: {:MESSAGE_REACTION_REMOVE, {map}, WSState.t()}
   @type message_reaction_remove_all :: {:MESSAGE_REACTION_REMOVE_ALL, {map}, WSState.t()}
   @type message_ack :: {:MESSAGE_ACK, {map}, WSState.t()}
+  @typedoc """
+  Dispatched when a user's presence is updated.
+  `old_presence` will be `nil` when the pre-update presence could not be fetched from the cache.
```
Comment (Kraigie) [inline lib/nostrum/consumer.ex]: Could you add a new line between the two sentences? ExDoc uses the first line to generate summaries, so we should try to keep it away from the rest of the provided info just for clarity.

File: lib/nostrum/consumer.ex [hunk 5]
```diff
@@ -123,16 +136,24 @@ defmodule Nostrum.Consumer do
...
   @type presence_update ::
           {:PRESENCE_UPDATE,
            {guild_id :: integer, old_presence :: map | nil, new_presence :: map}, WSState.t()}
   @type ready :: {:READY, {map}, WSState.t()}
   @type resumed :: {:RESUMED, {map}, WSState.t()}
   @type typing_start :: {:TYPING_START, {map}, WSState.t()}
   @type user_settings_update :: no_return
+  @typedoc """
+  Dispatched when a user is updated.
+  `old_user` will be `nil` when the pre-update user could not be fetched from the cache.
```
Comment (Kraigie) [inline lib/nostrum/consumer.ex]: Could you add a new line between the two sentences? ExDoc uses the first line to generate summaries, so we should try to keep it away from the rest of the provided info just for clarity.

**Comment by jchristgit** [comment]
Added the newlines & amended the commit.

**Comment by Kraigie** [comment]
Thanks!

**Pull request merged**